src/lib/canister_methods/pre_upgrade.ts

```diff
@@ -5,7 +5,7 @@ export function preUpgrade(
     callback: () => void | Promise<void>
 ): CanisterMethodInfo<[], Void> {
     const finalCallback = (...args: any[]) => {
-        executeMethod('preUpgrade', args, callback, [], Void, false);
+        callback();
```

Comment:
I think this can be reduced to just passing the callback in, get rid of the finalCallback concept